src/history/hs.c

```diff
@@ -219,6 +219,9 @@ __wt_hs_cursor_open(WT_SESSION_IMPL *session)
     /* History store cursors should always ignore tombstones. */
     F_SET(cursor, WT_CURSTD_IGNORE_TOMBSTONE);
 
+    /* Set the flag to stop creating snapshots for history store cursors. */
+    F_SET((WT_CURSOR_BTREE *)cursor, WT_CBT_NO_TXN);
```

Comment:
Does the cursor opened in `__hs_fixup_out_of_order_from_pos` need this too?

Reply:
Yes, that's a good point.

Reply:
Also we should flag it as

    /* History store cursors should always ignore tombstones. */
    F_SET(cursor, WT_CURSTD_IGNORE_TOMBSTONE);

Reply:
I don't think that matters. It's just only ever inserting things.